plugins/unifying/fu-unifying-bootloader.c

```diff
@@ -75,16 +77,53 @@ fu_unifying_bootloader_parse_requests (FuUnifyingBootloader *self, GBytes *fw, G
 		}
 		payload->addr = ((guint16) fu_unifying_buffer_read_uint8 (tmp + 0x03)) << 8;
 		payload->addr |= fu_unifying_buffer_read_uint8 (tmp + 0x05);
+		payload->cmd = FU_UNIFYING_BOOTLOADER_CMD_WRITE_RAM_BUFFER;
 
 		rec_type = fu_unifying_buffer_read_uint8 (tmp + 0x07);
 
-		/* record type of 0xFD indicates signature data */
-		if (rec_type == 0xFD) {
-			payload->cmd = FU_UNIFYING_BOOTLOADER_CMD_WRITE_SIGNATURE;
-		} else {
-			payload->cmd = FU_UNIFYING_BOOTLOADER_CMD_WRITE_RAM_BUFFER;
+		switch (rec_type) {
+			case 0x00: /* data */
+				break;
+			case 0x01: /* EOF */
+				exit = TRUE;
+				break;
+			case 0x03: /* start segment address */
+				/* this is used to specify the start address,
+				it is doesn't mater in this context so we can
+				safely ignore it */
+				continue;
+			case 0x04: /* extended linear address */
+				offset = ((guint16) fu_unifying_buffer_read_uint8 (tmp + 0x09)) << 8;
+				offset |= fu_unifying_buffer_read_uint8 (tmp + 0x11);
```

Comment:
not fu_unifying_buffer_read_uint16?

Reply:
Yes, it's big-endian. Maybe we could add a function for this?

Reply:
I mean, well, it shouldn't really matter if we parse it as be or le here because we just want to compare the value to 0. But I think parsing the memory properly makes the code more maintainable.

Reply:
I think we could/should add a `FuEndianType` argment to the shared `fu_firmware_strparse_uint16()` and also use `fu_firmware_strparse_uint8()` rather than `fu_unifying_buffer_read_uint8()`

Reply:
Could you elaborate a bit please :smile:? I am a bit lost in glib here. What is `FuEndianType` supposed to contain and how do I integrate it with `fu_firmware_strparse_uint16()`/`fu_unifying_buffer_read_uint8()`?

Reply:
Actually, I think you can just use `fu_firmware_strparse_uint16()` -- it's what's used in the other Intel HEX parsing code.